### DISCUSSION TITLE
Problem evaluating anonymous blocks

### DISCUSSION THREAD

**Discussion by cristobalinho** [mozman/ezdxf#863] [Q&A] [answered]
Hello everyone! I have some dynamic blocks on my dwg file, and when I save it as dxf, the name of the respective inserts appears with a new name corresponding to an anonymous block, so I can't identify which block corresponds to that insert. For example in the dwg file the dynamic block is called "windows" but in the dxf files the inserts and block corresponding to that block are named *U155 and the block "windows" still exists but with no Inserts associated. I wonder if there is a way to know the original name of the block corresponding to that Insert, that is, I need to obtain "windows" from the Insert "*U155" since this block was created from "windows". I know that dynamic blocks are not supported by dxf files, but I only need to know if there is a way to take back the original name. 

I also tried to setting the flags values of this anonymous blocks to 0 and something curious happened. Once I set all flags values to False, when I save the file and open again it shows that there are still some blocks with flags enabled,  but the number of flags decreased from 3 to 2. Here is the code I wrote.

```
anonymous_block_refs = msp.query('INSERT[name ? "^\*U.+"]')
A=[]
for i in anonymous_block_refs:
    if i.block().block.dxf.flags>0:
        A.append(i)

for i in A:
    if i.block().block.dxf.flags>0:
        for j in range(i.block().block.dxf.flags+1):
            i.block().block.set_flag_state(j+1,False)
            
doc.saveas('Test.dxf')
```
I hope I've been clear and thank you for your time!

**Comment by mozman** [comment, marked as answer]
Dynamic blocks are not documented in the DXF reference. To get help or information on this you need to use AutoCAD related forums.

**Reply by cristobalinho** [reply]
Thank you!
